go/parser/parser.go

```diff
@@ -228,6 +237,48 @@ func (p *parser) redeclared(ident *ast.Ident, prev *ast.Object, reason string) {
 	p.error(ident.Pos(), fmt.Sprintf("%s redeclared%s in this block%s", ident.Name, reason, prevDecl))
 }
 
+var (
+	goosSuffixes = []string{
+		"_dragonfly.go",
+		"_netbsd.go",
+		"_openbsd.go",
+		"_solaris.go",
+		"_freebsd.go",
+		"_nacl.go",
+		"_android.go",
+		"_plan9.go",
+		"_darwin.go",
+		"_linux.go",
+		"_windows.go",
+	}
+	goarchSuffixes = []string{
+		"_amd64.go",
+		"_amd64p32.go",
+		"_arm.go",
+		"_ppc64.go",
+		"_ppc64le.go",
+		"_386.go",
+	}
+)
+
+func isPlatformDependent(f *token.File) bool {
```

Comment:
If we're using the default context, how do we ever get more than one definition of a symbol with this code?

I have been pondering this issue for a while, as it also comes up in other contexts (for example, my godeps tool should really find dependencies from all architectures, which it currently does not).

Unfortunately I'm about out of time for the week now. I'll have a think over the weekend and when I get back on Monday.